.github/workflows/e2e-starter.yml

```diff
@@ -0,0 +1,71 @@
+# SPDX-License-Identifier: Apache-2.0
+
+name: E2E Starter
+
+on:
+  # run against every merge commit to 'main' and release branches
+  push:
+    branches:
+      - main
+      - release-*
+  # only run on PRs that touch certain regex paths
+  pull_request:
+    branches:
+      - main
+      - release-*
+    paths:
+      # note this should match the merging criteria in 'mergify.yml'
+      - '**.py'
+      - 'pyproject.toml'
+      - 'requirements**.txt'
+      - 'scripts/basic-workflow-tests.sh' # Used by this workflow
+      - 'scripts/test-data/**'
+      - '.github/workflows/e2e-starter.yml' # This workflow
+      - '.github/workflows/e2e-nvidia-t4-x1.yml' # Follow-on workflow
+
+env:
+  LC_ALL: en_US.UTF-8
+
+defaults:
+  run:
+    shell: bash
+
+permissions:
+  contents: read
+
+jobs:
+  e2e-status-checks:
+    if: ${{ github.repository == 'instructlab/instructlab' }}
+    permissions:
+      checks: read
+    uses: ./.github/workflows/status-checks.yml
+    with:
+      job_ids: >- # Space-separated job ids to wait on for status checks
+        DCO
+        actionlint
+        markdown-lint
+        shellcheck
+        spellcheck
+        lint-workflow-complete
+
+  e2e-starter:
+    needs:
+      - e2e-status-checks
+    runs-on: ubuntu-latest
+    steps:
+      - name: "Harden Runner"
+        uses: step-security/harden-runner@5c7944e73c4c2a096b17a9cb74d65b6c2bbafbde # v2.9.1
+        with:
+          egress-policy: audit # TODO: change to 'egress-policy: block' after couple of runs
+
+      - name: "Store Starter Metadata"
+        run: |
+          jq '{"ref": $ref, "number": $number, "event": .}' --arg ref "${{ github.ref }}" --arg number "${{ github.event.number || '0' }}" -- "$GITHUB_EVENT_PATH" > starter.json
+
+      - name: "Upload Starter Metadata artifact"
+        uses: actions/upload-artifact@50769540e7f4bd5e21e526ee35c689e35e0d6874 # v4.4.0
+        with:
+          name: e2e-starter
+          path: starter.json
+          if-no-files-found: error
+          retention-days: 5
```

Comment:
What happens if:
- PR created and fails on e2e
- 6 days later someone kicks off the e2e job again

I assume it will fail to find starter.json?

Reply:
Yes, if GitHub reclaimed the file. Of course, only a maintainer can re-run a failed job and it is likely the e2e will fail again if nothing changed in the PR.

90 days is max retention, so this is possible for any retention value + 1 day.

Reply:
Thanks!  Given it's just the one file, a longer retention might be worth it.